draft-trammell-taps-interface.md

```diff
@@ -1243,28 +1024,93 @@ connectionProperties := Connection.getProperties()
 Connection.setProperties()
 ~~~
 
-Connection properties may include:
+Connection properties include:
 
-* The status of the Connection, which can be one of the following: Establishing, Established, Closed. \[TASK: can connections be in a Closing state? (csp)]
+* The status of the Connection, which can be one of the following: Establishing, Established, Closing, or Closed.
 * Transport Features of the protocols that conform to the Required and Prohibited Transport Preferences, which might be selected by the transport system during Establishment. These features correspond to the properties given in {{transport-params}} and can only be queried.
 * Transport Features of the protocols that were selected, once the Connection has been established. These features correspond to the properties given in {{transport-params}} and can only be queried.
-* Protocol Properties of the protocols in use, once the Connection has been established. These properties correspond to the properties given {{protocol-props}} and can be set and queried.
-* Path Properties of the path(s) in use, once the Connection has been established. These properties can be derived from the local provisioning domain, measurements by the protocol stack, or other sources. They can only be queried.
+* Protocol Properties of the Protocol Stack in use (see {{protocol-props}} below). These can be set or queried. Certain specific procotol queries may be read-only, on a protocol and property specific basis.
+* Path Properties of the path(s) in use, once the Connection has been established. These  properties can be derived from the local provisioning domain, measurements by the protocol stack, or other sources. They can only be queried.
 
 {{appendix-specify-query-params}} gives a more detailed overview of the different types of properties that can be set and queried at different times.
 
+## Protocol Properties {#protocol-props}
+
+Protocol Properties represent the configuration of the selected Protocol Stacks
+backing a connection. Some properties apply generically across multiple
+transport protocols, while other properties only apply to specific protocols.
+The default settings of these properties will vary based on the specific
+protocols being used and the system's configuration.
+
+Note that Protocol Properties are also set during pre-establishment, as
+transport parameters, to preconfigure Protocol Stacks during establishment.
+
+Generic Protocol Properties include:
+
+* Timeout for aborting Connection: 
+  This numeric property specifies how long
+  to wait before aborting a Connection during establishment, or after a
+  connection has failed after establishment. It is given in seconds.
+
```

Comment:
"after a connection has failed after establishment" seems recursive, as such failure would be the result of the timeout. Suggestion: "...how long to wait before aborting a Connection during establishment, or before deciding that a connection has failed after establishment."

Reply:
Oops, missed this one. Fixed.